Automattic/jetpack
# VideoPress: Add `upload-jwt` endpoint

Introduce an endpoint and utility class to handle the VideoPress token.

Related with #26316 

#### Changes proposed in this Pull Request:

* Add new class `VideoPressToken`.
* Add new endpoint `upload-jwt`.
* Use `VideoPressToken` static methods to retrieve tokens in other parts.

#### Other information:


#### Jetpack product discussion


N/A

#### Does this pull request change what data or activity we track or use?


No.

#### Testing instructions:






_New Class_
* Build the plugin itself (`jetpack build plugins/jetpack`)
* Run `jetpack docker wp shell`
* Execute call for all methods:
* `Automattic\Jetpack\VideoPress\VideoPress_Rest_Api_V1_Token::upload_jwt();`
* `Automattic\Jetpack\VideoPress\VideoPressToken::videopress_playback_jwt();`
* `Automattic\Jetpack\VideoPress\VideoPressToken::videopress_onetime_upload_token();`
* `Automattic\Jetpack\VideoPress\VideoPressToken::videopress_upload_jwt();`

_Edit Video_

* Navigate to VideoPress dashboard.
* Edit poster for some video.
* Everything should work.
